verifier/src/main/java/com/okta/jwt/impl/OktaJWTClaimsVerifier.java

```diff
@@ -62,27 +71,44 @@ public void verify(JWTClaimsSet jwt, C context) throws BadJWTException {
         }
 
-        // validate audience
-        Object clientId;
-        String clientIdClaim;
-
-        if (TOKEN_TYPE_ID.equals(tokenType)) {
-            clientIdClaim = "aud";
-            clientId = jwt.getAudience();
-        } else {
-            clientIdClaim = "cid";
-            clientId = jwt.getClaims().get(clientIdClaim);
+        // Access Token Validation
+        if (TOKEN_TYPE_ACCESS.equals(tokenType)) {
+            List<String> resolvedAudience = jwt.getAudience();
+            // the expected audience MUST be in this list
+            if (CollectionUtils.isEmpty(resolvedAudience) || !resolvedAudience.contains(audience)) {
+                throw new BadJWTException(String.format("Failed to validate jwt string, invalid audience " +
+                        "claim 'aud', expected '%s', but found '%s'", audience, resolvedAudience));
+            }
         }
 
-        if (clientId instanceof Collection) {
-            Collection clientIdCollection = (Collection) clientId;
-            if (!clientIdCollection.contains(clientOrAudience)) {
-                throw new BadJWTException(String.format("Failed to validate jwt string, invalid clientId/audience " +
-                        "claim '%s', expected '%s', found '%s'", clientIdClaim, clientOrAudience, clientId));
+        // ID Token Validation
+        else if (TOKEN_TYPE_ID.equals(tokenType)) {
+
+            Assert.notNull(clientId, "An OAuth clientId must be specified when validating ID Tokens.");
+
+            List<String> resolvedAudience = jwt.getAudience();
+            // the expected audience MUST be in this list
+            if (CollectionUtils.isEmpty(resolvedAudience) || !resolvedAudience.contains(clientId)) {
+                throw new BadJWTException(String.format("Failed to validate jwt string, invalid clientId found in " +
+                        "claim 'aud', expected '%s', but found '%s'", clientId, resolvedAudience));
             }
-        } else if (!clientOrAudience.equals(clientId)) {
-            throw new BadJWTException(String.format("Failed to validate jwt string, invalid clientId/audience " +
-                    "claim '%s', expected '%s', found '%s'", clientIdClaim, clientOrAudience, clientId));
+
+            // validate nonce
+            Object resolvedNonce = jwt.getClaim("nonce");
+            if ( nonce != null && !nonce.equals(resolvedNonce)) {
+                throw new BadJWTException(String.format("Invalid nonce found in ID Token, expected '%s', but found '%s'", nonce, resolvedNonce));
+            }
+
+            String subject = jwt.getSubject();
+            if (subject == null) {
+                throw new BadJWTException("Invalid ID Token, missing subject claim ('sub')");
+            }
+
+        }
+
+        // Unknown Token Type
+        else {
+            throw new BadJWTException(String.format("Unknown token type: '%s'", tokenType));
         }
     }
-}
+}
```

Comment:
I'm a +0 on this one, some people like them some people don't. We don't have a standard code style one way or the other